src/ape/api/providers.py

```diff
@@ -136,10 +136,24 @@ def decode(cls, data: dict) -> "ReceiptAPI":
         ...
 
 
+@abstractdataclass
+class BlockGasAPI:
+    raw_data: Dict
+
+    @property
+    @abstractmethod
+    def base_fee(self) -> Optional[int]:
```

Comment:
I think `base_fee` might be lower level, my idea was thinking about other properties that might be a bit higher level and still very relevant

Although specifically, I think knowing `base_fee` would be very useful assuming EIP1559 is active and we want to be able to price gas costs for a new transaction

Reply:
Yeah we are actually calling this somewhere. Should we make subclasses or something?

Reply:
This is half the reason I need the `BlockAPI` to exist right now! So to be able to access this easily would be nice but I will settle for using dict-key access if you prefer.

Reply:
Lol I don't know what to do here!

I think we should do something that is easy to iterate on, whatever is, and add stuff on a per-needs basis.

I will think on it.

Reply:
We have `base_fee` on the `ProviderAPI` too, but it just raised `NotImplemented` by default. This is the same idea kind of except it returns `None`, which is a little nicer. 

Maybe raising `NotImplented` would be better though?

Reply:
I think returning `None` is fine, it is a little awkward way to abstract, because old style only has one parameter, and new style has two. Let's just do our best here and see how it feels?

Reply:
Yeah maybe returning `None` for everything is the best solution...

1. It is relaxed, like `dict`, in that those keys don't have to exist.
2. Any new keys that end up be needed can always be added and kept optional. Chains that to use them know should be there.